monitoring/ceph-mixin/prometheus_alerts.libsonnet

```diff
@@ -0,0 +1,695 @@
+            summary: 'OSD(s) too full for backfill operations%(cluster)s' % $.MultiClusterSummary(),
+            description: "An OSD has reached it's BACKFILL FULL threshold. This will prevent rebalance operations completing for some pools. Check the current capacity utilisation with 'ceph df'  To resolve, either add capacity to the cluster, or delete unwanted data ",
+          },
+        },
+        {
+          alert: 'CephOSDTooManyRepairs',
+          'for': '30s',
+          expr: 'ceph_health_detail{name="OSD_TOO_MANY_REPAIRS"} == 1',
+          labels: { severity: 'warning', type: 'ceph_default' },
+          annotations: {
+            documentation: 'https://docs.ceph.com/en/latest/rados/operations/health-checks#osd-too-many-repairs',
+            summary: 'OSD has hit a high number of read errors%(cluster)s' % $.MultiClusterSummary(),
+            description: 'Reads from an OSD have used a secondary PG to return data to the client, indicating a potential failing disk. ',
+          },
+        },
+        {
+          alert: 'CephOSDTimeoutsPublicNetwork',
+          'for': '1m',
+          expr: 'ceph_health_detail{name="OSD_SLOW_PING_TIME_FRONT"} == 1',
+          labels: { severity: 'warning', type: 'ceph_default' },
+          annotations: {
+            summary: 'Network issues delaying OSD heartbeats (public network)%(cluster)s' % $.MultiClusterSummary(),
+            description: "OSD heartbeats on the cluster's 'public' network (frontend) are running slow. Investigate the network for any latency issues on this subnet. Use 'ceph health detail' to show the affected OSDs. ",
+          },
+        },
+        {
+          alert: 'CephOSDTimeoutsClusterNetwork',
+          'for': '1m',
+          expr: 'ceph_health_detail{name="OSD_SLOW_PING_TIME_BACK"} == 1',
+          labels: { severity: 'warning', type: 'ceph_default' },
+          annotations: {
+            summary: 'Network issues delaying OSD heartbeats (cluster network)%(cluster)s' % $.MultiClusterSummary(),
+            description: "OSD heartbeats on the cluster's 'cluster' network (backend) are running slow. Investigate the network for any latency issues on this subnet. Use 'ceph health detail' to show the affected OSDs. ",
+          },
+        },
+        {
+          alert: 'CephOSDInternalDiskSizeMismatch',
+          'for': '1m',
+          expr: 'ceph_health_detail{name="BLUESTORE_DISK_SIZE_MISMATCH"} == 1',
+          labels: { severity: 'warning', type: 'ceph_default' },
+          annotations: {
+            documentation: 'https://docs.ceph.com/en/latest/rados/operations/health-checks#bluestore-disk-size-mismatch',
+            summary: 'OSD size inconsistency error%(cluster)s' % $.MultiClusterSummary(),
+            description: "One or more OSDs have an internal inconsistency between the size of the physical device and it's metadata. This could lead to the OSD(s) crashing in future. You should redeploy the effected OSDs. ",
```

Comment:
s/it's//
"This could lead to OSDs crashing in the future.  You should look for hardware errors and redeploy affected OSDs"